libc/watcom/syscall/fmemalloc.c

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+*
+*                            Open Watcom Project
+*
+* Copyright (c) 2002-2019 The Open Watcom Contributors. All Rights Reserved.
+*    Portions Copyright (c) 1983-2002 Sybase, Inc. All Rights Reserved.
+*
+*  ========================================================================
+*
+*    This file contains Original Code and/or Modifications of Original
+*    Code as defined in and that are subject to the Sybase Open Watcom
+*    Public License version 1.0 (the 'License'). You may not use this file
+*    except in compliance with the License. BY USING THIS FILE YOU AGREE TO
+*    ALL TERMS AND CONDITIONS OF THE LICENSE. A copy of the License is
+*    provided with the Original Code and Modifications, and is also
+*    available at www.sybase.com/developer/opensource.
+*
+*    The Original Code and all software distributed under the License are
+*    distributed on an 'AS IS' basis, WITHOUT WARRANTY OF ANY KIND, EITHER
+*    EXPRESS OR IMPLIED, AND SYBASE AND ALL CONTRIBUTORS HEREBY DISCLAIM
+*    ALL SUCH WARRANTIES, INCLUDING WITHOUT LIMITATION, ANY WARRANTIES OF
+*    MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, QUIET ENJOYMENT OR
+*    NON-INFRINGEMENT. Please see the License for the specific language
+*    governing rights and limitations under the License.
+*
+*  ========================================================================
+*
+* Description:  ELKS _fmemalloc() system call.
+*
+****************************************************************************/
+
+
+#include <malloc.h>
+#include "watcom/syselks.h"
+
+int _fmemalloc( int __paras, unsigned short *__pseg )
+{
+    sys_set_ptr_seg(__pseg);
+    sys_setseg(__pseg);
+    syscall_res res = sys_call2( SYS_fmemalloc, __paras, __pseg);
```

Comment:
I thought the `wdis` code looked correct, but we should write `(unsigned)__pseg` to ensure that only the offset portion of `__pseg` is loaded into registers by the `sys_call2` macro. See sbrk.c for example.